### ISSUE TITLE
Serial device connected to Arduino Uno TX/RX pins (or others digital pins)

### ISSUE THREAD

**Issue by benmarsh** [rwaldron/johnny-five#247]
I have a Adafruit Ultimate GPS module which has serial TX/RX pins. I can see data by hooking it up to the Uno's TX/RX pins, loading a blank Arduino sketch and opening serial monitor. With the Firmata sketch loaded data doesn't show.

Is there a way to make the board's serial port visible to Firmata, so I can read the GPS in Johnny-Five - or is it limited to Firmata-only communication?

If not, could I use other pins on the board to read the serial data?

Alternatively, the Arduino Mega has a number of serial ports on-board. Could Johnny-Five (or Firmata) potentially communicate with the GPS over one of those?

**Comment by kirkstrobeck** [comment]
+1 on johnny five gps

**Comment by dtex** [comment]
Now that we have serial for firmata can we remove the "On Hold" label from this?

**Comment by rwaldron** [comment]
Sure

**Comment by dtex** [comment]
I've got J5 working with the Adafruit Ultimate GPS module on [this branch](https://github.com/dtex/johnny-five/tree/gps). 

````
var five = require("../lib/johnny-five.js");
var board = new five.Board();

board.on("ready", function() {
  var gps = new five.GPS([11, 10]);
  gps.on("change", function(data) {
    console.log(data);
  });
});

// { 
//    gpsFixData: '$GPGGA,062753.000,2940.0109,N,09527.0349,W,2,10,0.87,22.9,M,-23.7,M,0000,0000*64',
//    UTCtime: '062753.000',
//    lat: '29.6668',
//    lng: '-95.4505,
//    alt: '22.9' 
// }
````

The sensitivity is really high and the accuracy is within 10 feet. There is soooo much data we can expose that it really seems like overkill. I just have it emitting the minimum but it's easy enough to expose more.

Todo:

1. Get my hands on some more GPS breakouts so I can figure out what to put into the default methods and what needs to be in separate controllers/devices.
2. Write some tests.
3. Figure out how to write configuration commands to the device(s). I'm probably just building my strings wrong.

**Comment by rwaldron** [comment]
This is a fantastic start!!

**Comment by dtex** [comment]
Landed in 0.9.32
